ql/src/java/org/apache/hadoop/hive/ql/parse/PTFTranslator.java

```diff
@@ -142,6 +143,19 @@ public PTFDesc translate(WindowingSpec wdwSpec, SemanticAnalyzer semAly, HiveCon
       UnparseTranslator unparseT)
       throws SemanticException {
     init(semAly, hCfg, inputRR, unparseT);
+    for (int i = 0; i < wdwSpec.getWindowExpressions().size(); ++i) {
```

Comment:
Should we do this rewriting via Calcite rule instead of adding this custom logic over here?